### ISSUE TITLE
micro tasks

### ISSUE THREAD

**Issue by james2432** [hotosm/osm-tasking-manager2#683]
E.g.
Each tile in a task would have 
-Draw roads
-import address data
-draw water sources
-trace buildings

And each one could be checked off and reviewed

**Comment by bgirardot** [comment]
Hi @james2432 our friends at MSF created an app a lot like this. In stead of actually mapping you just record if you see something of that type.

http://mapswipe.org/

There is also MapRoulette which support similar functionality in a web browser.

Cheers,
Blake

**Comment by james2432** [comment]
Maybe you misunderstood. The current workflow is work->done->validate.

What I wanted was subclassing of work that needed to be done that could be configured and checked off in each tile. Once all the tasks were done in a tile, then that tile becomes "done"

**Comment by bgirardot** [comment]
This sounds exactly like http://www.maproulette.org/ but there is not tasking above the micro editing level.

**Comment by althio** [comment]
This is similar to a few ideas around 'checkboxes' or 'popup' or 'template/form'.
My understanding is:
- potential duplicate of #421 ( #14 )
- related to https://github.com/hotosm/osm-tasking-manager2/issues/427 #172

**Comment by james2432** [comment]
It may be this: https://github.com/hotosm/osm-tasking-manager2/issues/172

If by form he means in each tile there is an html form to submit the completeness of a tile.
I.e
Your instructions for a task is to trace roads and trace buildings

So once someone finishes roads they tick off roads and that tile is 50% complete. When they finish buildings it becomes done

**Comment by bgirardot** [comment]
@james2432 I think I get what you are talking about now.

My initial reaction is that we have a very difficult time now getting people to understand the difference between "Stop Mapping" and "Done" adding in more complexity does not seem like a direction we want to go. But I am reopening this so it stays in the mix as it is directly impacting a major design issue in TM3.0

**Comment by james2432** [comment]
Its just it's easier to see what needs to be accomplished in a tile i.e. someone that hates drawing buildings and just wants to draw roads would look for tiles that may not have them "completed". It would also more accurately depict the completeness of the task itself as waiting for everything to be complete doesnt show the reality as the tile may be 99.9% complete, but until it's marked as "done" it doesnt count towards the overall progress

**Comment by bgirardot** [comment]
@james2432 Some of this could be addressed by better project creation, smaller task squares, more one type of feature only mapping, roads, buildings, waterways, etc. We get other feedback that says multiple object types in a project make them harder to complete often.

**Comment by james2432** [comment]
Sometimes its the same project aka building/address imports for same geo location, thus you would have to create same project twice

**Comment by bgirardot** [comment]
Yes, you would create exactly the same project twice. If the instructions are clear you can map them at the same time. Unless your specific use case is mapping buildings while doing an address import, that you might have to as one project. Buildings and roads are easy to separate and map simultaneously. Buildings and addresses probably not.
